pysatMadrigal/instruments/gnss_tec.py

```diff
@@ -0,0 +1,215 @@
+
+    return
+
+
+def download(date_array, tag='', sat_id='', data_path=None, user=None,
+             password=None):
+    """Downloads data from Madrigal.
+
+    Parameters
+    ----------
+    date_array : array-like
+        list of datetimes to download data for. The sequence of dates need not
+        be contiguous.
+    tag : string
+        Tag identifier used for particular dataset. This input is provided by
+        pysat. (default='')
+    sat_id : string
+        Satellite ID string identifier used for particular dataset. This input
+        is provided by pysat. (default='')
+    data_path : string
+        Path to directory to download data to. (default=None)
+    user : string
+        User string input used for download. Provided by user and passed via
+        pysat. If an account is required for dowloads this routine here must
+        error if user not supplied. (default=None)
+    password : string
+        Password for data download. (default=None)
+
+    Note
+    ----
+    The user's names should be provided in field user. Anthea Coster should
+    be entered as Anthea+Coster
+
+    The password field should be the user's email address. These parameters
+    are passed to Madrigal when downloading.
+
+    The affiliation field is set to pysat to enable tracking of pysat
+    downloads.
+
+    """
+    mad_meth.download(date_array, inst_code=str(madrigal_inst_code),
+                      kindat=str(madrigal_tag[sat_id][tag]),
+                      data_path=data_path, user=user, password=password)
+    return
+
+
+def load(fnames, tag=None, sat_id=None):
+    """ Routine to load the GNSS TEC data
+
+    Parameters
+    -----------
+    fnames : list
+        List of filenames
+    tag : string or NoneType
+        tag name used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself. (default=None)
+    sat_id : string or NoneType
+        Satellite ID used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself. (default=None)
+
+    Returns
+    --------
+    data : xarray.Dataset
+        Object containing satellite data
+    meta : pysat.Meta
+        Object containing metadata such as column names and units
+
+    """
+    # Define the xarray coordinate dimensions (apart from time)
+    xcoords = {'vtec': {('time', 'gdlat', 'glon', 'gdalt', 'kindat', 'kinst'):
+                        ['tec', 'dtec'],
+                        ('time',): ['year', 'month', 'day', 'hour', 'min',
+                                    'sec', 'ut1_unix', 'ut2_unix', 'recno']},}
+
+    # Load the specified data
+    data, meta = mad_meth.load(fnames, tag, sat_id, xarray_coords=xcoords[tag])
+
+    # Squeeze the kindat and kinst 'coordinates', but keep them as floats
+    data = data.squeeze(dim=['kindat', 'kinst'])
+
+    # Fix the units for tec and dtec
+    if tag == 'vtec':
+        meta['tec'] = {'units': 'TECU'}
```

Comment:
I messed up my generalization comment earlier. It works but not for this particular method.

The labels in Meta are set by the madrigal methods load. Technically the suggestion is more robust but we also control what the madrigal load returns.

Reply:
Things were posted out of order, suggestion referred to is a couple boxes below. with meta.units_label.